src/library_manager/lib_notification.c

```diff
@@ -0,0 +1,117 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Jaroslaw Stelter <jaroslaw.stelter@intel.com>
+//         Pawel Dobrowolski <pawelx.dobrowolski@intel.com>
+//
+
+/*
+ * 3-rd party modules do not have access to component device runtime data.
+ * System service API used by these modules require notification handling
+ * based on base FW message management. Therefore this code expose notification
+ * calls aligned with cAVS/ACE System service API requirements.
+ */
+
+#include <sof/list.h>
+#include <sof/ipc/msg.h>
+#include <sof/lib/memory.h>
+#include <sof/lib_manager.h>
+
+struct ipc_msg *lib_notif_msg_init(uint32_t header, uint32_t size)
+{
+	struct ipc_msg *msg = NULL;
+	struct ipc_lib_msg *msg_pool_elem;
+	struct ext_library *_ext_lib = ext_lib_get();
+	struct ipc_lib_msg *lib_notif = _ext_lib->lib_notif_pool;
+
+	if (lib_notif) {
+		/* If list not empty search for free notification handle */
+		if (!list_is_empty(&lib_notif->list)) {
+			struct list_item *list_elem;
+			/* Search for not used message handle */
+			list_for_item(list_elem, &lib_notif->list) {
+				msg_pool_elem = container_of(list_elem, struct ipc_lib_msg, list);
+				if (msg_pool_elem->msg != 0) {
+					if (list_is_empty(&msg_pool_elem->msg->list)) {
+						msg = msg_pool_elem->msg;
+						break;
+					}
+				}
+			}
+		}
+	}
+
+	if (!msg) {
+		k_spinlock_key_t key;
+		/* No free element or list empty, create new handle  */
+		if (_ext_lib->lib_notif_count > LIB_MANAGER_LIB_NOTIX_MAX_COUNT)
+			return NULL;
```

Comment:
we should emit some trace here to tell the user they have too many modules in the lib